app/jobs/deposit_job.rb

```diff
@@ -76,4 +76,32 @@ def build_file_metadata(blobs)
   def filename(key)
     ActiveStorage::Blob.service.path_for(key)
   end
+
+  def update_dro_with_existing_file_identifiers(request_dro, druid)
+    request_dro.new(structural: request_dro.structural.new(contains: existing_structural_for(druid)))
+  end
+
+  def metadata_only?(work_version)
+    previous_version = previous_version_for(work_version)
+    return false unless previous_version
+
+    blob_map_for(work_version) == blob_map_for(previous_version)
+  end
+
+  def previous_version_for(work_version)
+    work_version.work.work_versions.find { |check_work_version| check_work_version.version = work_version.version - 1 }
```

Comment:
How about moving this to the WorkVersion class?

Reply:
Done.